### PR TITLE
Also print a memory usage table

### PR DESCRIPTION
**Pull request by abelsiqueira**
By default, a second table with the number of allocations and memory usage will be printed as well. This can be suppressed with the flag --hide-memory.

Example output:

```
$ ~/.julia/bin/benchpkgtable AirspeedVelocity --rev v0.4.4,v0.4.3
[ Info: Loading results from ./results_AirspeedVelocity@v0.4.4.json
[ Info: Flattening results.
[ Info: Loading results from ./results_AirspeedVelocity@v0.4.3.json
[ Info: Flattening results.
|                   | v0.4.4          | v0.4.3          |
|:------------------|:---------------:|:---------------:|
| main/random_sleep | 0.532 ± 0.44 s  | 0.404 ± 0.28 s  |
| time_to_load      | 0.338 ± 0.002 s | 0.34 ± 0.0053 s |

|                   | v0.4.4                  | v0.4.3                  |
|:------------------|:-----------------------:|:-----------------------:|
| main/random_sleep | 4  allocs: 0.112 kB     | 4  allocs: 0.112 kB     |
| time_to_load      | 0.214 M allocs: 14.5 MB | 0.217 M allocs: 14.6 MB |
```

Partially implements #8.

### CODE REVIEW

**Review by MilesCranmer — CHANGES_REQUESTED** [review]
Nice!! Looks good. I was wondering if you could change the CLI flags to be as follows instead?

```bash
benchpkgtable --mode=memory
```

where the default `--mode=time`. And `--mode=time,memory` would be like what you have now.

Since you can't output two tables at once this will make it easier to partition the `benchmark_pr.yml` command so the memory table can be collapsed by default.

**Comment by MilesCranmer** [comment]
Also I'm not sure why but the columns seem to be `median` and `memory` in the ratio column?

Screenshot 2024-02-22 at 11 41 36

File: src/Utils.jl [hunk 1]
```diff
@@ -15,15 +15,27 @@ function get_spec_str(spec::PackageSpec)
...
     return unit, unit_name
 end
 
+function get_reasonable_time_unit(quantities::AbstractArray)
+    units = [(1.0, "ns"), (1e-3, "μs"), (1e-6, "ms"), (1e-9, "s"), (1e-9 / 3600, "h")]
+    return get_reasonable_unit(median(quantities), units)
+end
+
+function get_reasonable_memory_unit(memory)
+    units = [(1.0, "B"), (1e-3, "kB"), (1e-6, "MB"), (1e-9, "GB")]
```
Comment (MilesCranmer) [inline src/Utils.jl:30]: I think kB should be `1/1024` rather than `1/1000`, right?

Comment (MilesCranmer) [reply src/Utils.jl]: I fixed this but the test needs to be updated too which seems trickier.

**Review by MilesCranmer — APPROVED** [review]

**Comment by MilesCranmer** [comment]
Thanks @abelsiqueira!!

**Comment by abelsiqueira** [comment]
Thanks for the updates and the merge, @MilesCranmer.

**Pull request merged**